File: scripts/optimization/cache-warmup.jl
#!/usr/bin/env julia
using JSON, Dates

println("\n" * "="^70)
println("PHASE 5: PILLAR 1 — CACHE WARMUP")
println("="^70 * "\n")

files = [("SOUL.md", "SOUL.md"), ("IDENTITY.md", "IDENTITY.md"), ("AGENTS.md", "AGENTS.md"), ("MEMORY.md", "MEMORY.md")]
total = 0
bundle_size = 0

println("Loading context files...")
for (name, path) in files
    if isfile(path)
        c = read(path, String)
        sz = length(c)
        toks = floor(Int, length(split(c)) * 1.3)
        total += toks
        bundle_size += sz
        println("  ✓ $name: $toks tokens")
    end
end

println("\n✅ Bundle ready: $total tokens, $bundle_size chars")

warmup = Dict("timestamp" => string(now()), "total_tokens" => total, "status" => "ready")
mkpath("data/metrics")
open("data/metrics/cache-warmup-state.json", "w") do f
    write(f, JSON.json(warmup, 2))
end

println("✅ Saved warmup state\n" * "="^70 * "\n")

File: scripts/optimization/cache-warmup.py
#!/usr/bin/env python3
"""
Phase 5: Pillar 1 — Cache Warmup (Python version)
"""
import json
import os
from datetime import datetime

print("\n" + "="*70)
print("PHASE 5: PILLAR 1 — CACHE WARMUP")
print("="*70 + "\n")

files = [
    "SOUL.md",
    "IDENTITY.md",
    "AGENTS.md",
    "MEMORY.md"
]

total_tokens = 0
total_size = 0

print("Loading context files...")
for fname in files:
    if os.path.exists(fname):
        with open(fname) as f:
            content = f.read()
            size = len(content)
            tokens = len(content.split()) * int(1.3)  # Rough estimate
            total_tokens += tokens
            total_size += size
            print(f"  ✓ {fname}: {tokens} tokens")
    else:
        print(f"  ✗ {fname}: Not found")

print(f"\n✅ Bundle ready: {total_tokens} tokens, {total_size} chars")

warmup = {
    "timestamp": datetime.now().isoformat(),
    "total_tokens": total_tokens,
    "total_size": total_size,
    "files": files,
    "status": "ready"
}

os.makedirs("data/metrics", exist_ok=True)
with open("data/metrics/cache-warmup-state.json", "w") as f:
    json.dump(warmup, f, indent=2)

print("✅ Saved warmup state\n" + "="*70 + "\n")
